superset/connectors/sqla/models.py

```diff
@@ -130,8 +131,9 @@ def get_timestamp_expression(self, time_grain):
                 elif pdf == 'epoch_ms':
                     expr = db_spec.epoch_ms_to_dttm().format(col=expr)
             grain = self.table.database.grains_dict().get(time_grain)
-            literal = grain.function if grain else '{col}'
-            literal = expr.format(col=expr)
+            if grain:
+                literal = grain.function
+        literal = expr.format(col=expr)
```

Comment:
Isn't this line always overriding the one before?

Reply:
Yeah, I think this should be

```python
literal = literal.format(col=expr)
```

Reply:
See https://github.com/apache/incubator-superset/pull/4892

Reply:
Yeah I think we're both wrong on this one :( , I'm going to clean this up a bit more and write many tests covering the whole function.